draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -1581,75 +1581,141 @@ Steps:
 26. return (x, y)
 ~~~
 
-## Mappings for twisted Edwards curves
-
-### Elligator 2 Method {#ell2edwards}
-
-Twisted Edwards curves and Montgomery curves are closely related: every
-twisted Edwards curve is birationally equivalent to a Montgomery curve
-({{BBJLP08}}, Theorem 3.2). To hash to a twisted Edwards curve, hash
-to the equivalent Montgomery curve and evaluate the rational map to
-obtain a point on the twisted Edwards curve.
-
-For a twisted Edwards curve given by a * x^2 + y^2 = 1 + d * x^2 * y^2,
-first compute A and B, the parameters of the equivalent Montgomery curve,
-as follows:
-
-- A = (a + 2) / 2
+## Mappings for Edwards curves
+
+Edwards curves are closely related to Montgomery
+curves ({{montgomery}}): every Edwards curve is birationally equivalent
+to a Montgomery curve ({{BBJLP08}}, Theorem 3.2).
+This equivalence yields an efficient way of hashing to an Edwards curve:
+first, hash to the equivalent Montgomery curve, then transform the
+result into a point on the Edwards curve by evaluating a pair of rational
+functions called a birational map.
```

Comment:
here is an inaccuracy.
the `bi` prefix on birationality refers to the fact that there are two maps one from X -> Y and another from Y -> X .
and the 'bi' prefix is not related to the dimension of the varieties (the number of coordinates to represent a point).

Reply:
You're right, the wording is confusing here. I'll fix.

Just to make sure we're on the same page: X -> Y is a birational map just if it's a rational map *and* if the inverse map Y -> X also exists and is rational. In other words, it is correct to call the map M -> E birational.

So the objection here is that the wording seems to imply that "bi" and dimensionality are related, which of course as you say is not true.

Reply:
> Just to make sure we're on the same page: X -> Y is a birational map just if it's a rational map and if the inverse map Y -> X also exists and is rational. In other words, it is correct to call the map M -> E birational.

^ This is true.

Yes, the objection is in this part: 

> by evaluating **a pair** of rational functions called a birational map

In fact, we only evaluate **one** birational map, the one that goes from M -> E.